improver_tests/utilities/test_save.py

```diff
@@ -234,6 +234,19 @@ def test_error_unknown_units(self):
         with self.assertRaisesRegex(ValueError, msg):
             save_netcdf(no_units_cube, self.filepath)
 
+    def test_add_least_significant_digit(self):
+        """Test bitshaving adds correct metadata"""
+        save_netcdf(self.cube, self.filepath, least_significant_digit=2)
+        cube = load_cube(self.filepath)
+        self.assertEqual(cube.attributes["least_significant_digit"], 2)
+
+    def test_update_least_significant_digit(self):
+        """Test bitshaving updates metadata correctly if already present"""
+        self.cube.attributes["least_significant_digit"] = 0
+        save_netcdf(self.cube, self.filepath, least_significant_digit=2)
+        cube = load_cube(self.filepath)
+        self.assertEqual(cube.attributes["least_significant_digit"], 2)
+
```

Comment:
If I understand this correctly, an attribute "least_significant_digit" should be discarded on save if that kwarg is not set. Can you add a test for this too?